pkg/lakom/resolvetag/resolvetag_suite_test.go

```diff
@@ -5,13 +5,173 @@
 package resolvetag_test
 
 import (
+	"crypto"
+	"crypto/ecdsa"
+	"encoding/base64"
+	"fmt"
+	"log"
+	"net/http/httptest"
+	"net/url"
+	"os"
 	"testing"
 
+	mockclient "github.com/gardener/gardener/third_party/mock/controller-runtime/client"
+	mockmanager "github.com/gardener/gardener/third_party/mock/controller-runtime/manager"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/registry"
+	registryv1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/random"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	"github.com/sigstore/cosign/v2/pkg/cosign"
+	"github.com/sigstore/cosign/v2/pkg/oci/mutate"
+	ociRemote "github.com/sigstore/cosign/v2/pkg/oci/remote"
+	"github.com/sigstore/cosign/v2/pkg/oci/signed"
+	"github.com/sigstore/cosign/v2/pkg/oci/static"
+	"github.com/sigstore/sigstore/pkg/signature"
+	"go.uber.org/mock/gomock"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var (
+	// FullRef use the digest instead of the tag.
+	signedImageFullRef   string
+	unsignedImageFullRef string
+
+	// TagRef use the tag instead of the digest for referencing the artifact.
+	signedImageTagRef      string
+	unsignedImageTagRef    string
+	nonExistentImageTagRef string
+
+	scheme    *runtime.Scheme
+	ctrl      *gomock.Controller
+	mgr       *mockmanager.MockManager
+	apiReader *mockclient.MockReader
+)
+
+const (
+	signedImageTag   = "signed"
+	unsignedImageTag = "unsigned"
 )
 
 func TestCMD(t *testing.T) {
 	RegisterFailHandler(Fail)
 	RunSpecs(t, "ResolveTag Suite")
 }
+
+var _ = BeforeSuite(func() {
+	scheme = runtime.NewScheme()
+	err := corev1.AddToScheme(scheme)
+	Expect(err).ToNot(HaveOccurred())
+
+	dirPath, err := os.MkdirTemp("", "verifysignature_test")
+	Expect(err).ToNot(HaveOccurred())
+	DeferCleanup(func() {
+		err := os.RemoveAll(dirPath)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	// Tests rely on a fake registry
+	registryURL, server := startRegistry()
+	DeferCleanup(func() {
+		server.Close()
+	})
+
+	signedImage, signedImageRef, err := createTestImage(registryURL, signedImageTag)
+	Expect(err).ToNot(HaveOccurred())
+	_, unsignedImageRef, err := createTestImage(registryURL, unsignedImageTag)
+	Expect(err).ToNot(HaveOccurred())
+
+	signedImageFullRef = signedImageRef.Name()
+	unsignedImageFullRef = unsignedImageRef.Name()
+
+	signedImageTagRef = signedImageRef.Context().Tag(signedImageTag).String()
+	unsignedImageTagRef = unsignedImageRef.Context().Tag(unsignedImageTag).String()
+
+	nonExistentImageTagRef = fmt.Sprintf("%s:nonexistant", signedImageRef.Context().Name())
+
+	privateKey, err := cosign.GeneratePrivateKey()
+	Expect(err).ToNot(HaveOccurred())
+
+	err = signImage(signedImage, signedImageFullRef, privateKey)
+	Expect(err).ToNot(HaveOccurred())
```

Comment:
```suggestion
	Expect(signImage(signedImage, signedImageFullRef, privateKey)).To(Succeed())
```

The other pattern is used when the function returns more than one result.